GTG/gtk/browser/adaptive_button.py

```diff
@@ -0,0 +1,283 @@
+                          ci.widget, ci.minimum_height, ci.natural_height)
+        self._determine_and_save_active_child()
+
+        if self._spammy_debug:
+            log.debug("(width=%r) return minimum=%r, natural=%r",
+                      width, minimum, natural)
+        return (minimum, natural)
+
+    def do_get_preferred_height(self) -> Tuple[int, int]:
+        # As suggested by the GtkContainer docs
+        min_width, nat_width = self.get_preferred_width()
+        minimum, natural = self.get_preferred_height_for_width(min_width)
+
+        if self._spammy_debug:
+            log.debug("return minimum=%r, natural=%r", minimum, natural)
+        return (minimum, natural)
+
+    def do_get_preferred_width_for_height(self, height: int
+                                          ) -> Tuple[int, int]:
+        # As suggested by the GtkContainer docs
+        minimum, natural = self.get_preferred_width()
+
+        if self._spammy_debug:
+            log.debug("(height=%r) return minimum=%r, natural=%r",
+                      height, minimum, natural)
+        return (minimum, natural)
+
+    def do_draw(self, cr):
+        draw_child = None
+        allocation = self.get_allocation()
+        width = allocation.width
+        if self._spammy_debug:
+            log.debug("width=%r, children=%r", width,
+                      list(map(lambda ci: ci.minimum_width, self._children)))
+        draw_child = self._child_to_draw
+        if draw_child is not None:
+            draw_child.widget.draw(cr)
+        else:
+            log.debug("Found no suitable child")
+
+    def do_realize(self):
+        self.set_realized(True)
+
+        allocation = self.get_allocation()
+        attributes = Gdk.WindowAttr()
+        attributes.window_type = Gdk.WindowType.CHILD
+        attributes.x = allocation.x
+        attributes.y = allocation.y
+        attributes.width = allocation.width
+        attributes.height = allocation.height
+        attributes.wclass = Gdk.WindowWindowClass.INPUT_OUTPUT
+        attributes.visual = self.get_visual()
+        attributes.event_mask = self.get_events() | Gdk.EventMask.EXPOSURE_MASK
+
+        window = Gdk.Window.new(self.get_parent_window(), attributes,
+                                Gdk.WindowAttributesType.VISUAL |
+                                Gdk.WindowAttributesType.X |
+                                Gdk.WindowAttributesType.Y)
+        self.register_window(window)
+        self.set_window(window)
+
+    def do_size_allocate(self, allocation: Gdk.Rectangle):
+        if self._spammy_debug:
+            log.debug("allocation={x=%r, y=%r, w=%r, h=%r}",
+                      allocation.x, allocation.y,
+                      allocation.width, allocation.height)
+
+        self.set_allocation(allocation)
+        for ci in self._children:
+            ci.widget.size_allocate(allocation)
+        self._determine_and_save_active_child()
+
+    # Gtk.Container implementation
```

Comment:
This is a sort of "title comment", right? I find useful to surround these with lines, or at least one line after.

```
# -----------------------------------------------------------------------------
# Gtk.Container implementation
# -----------------------------------------------------------------------------
```

Otherwise it looks like commented code at first glance, or some regular comment about the code

Reply:
Yeah, a section title comment.. I applied the suggestion for now.